### PR TITLE
Blog for pdf preview and download

### PR DESCRIPTION
**Pull request by vivek-dhayalan**

### CODE REVIEW

**Review by prasanth-reddy-wavemaker-com — CHANGES_REQUESTED** [review]
We should show how design the prefab with the PDF file url as a bindable parameter so that the mobile app developer who is using the prefab can program where the pdf file itself comes from.

We should also mention that the service serving the PDF file should be properly configured so that the preview works. For example, the service should not be setting 'Content-disposition' header to attachment.

Did we check if the preview works with pdf files that are password protected?

File: learn/react-native/pdf-preview-and-download.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: "PDF preview"
+id: "pdf-preview-and-download"
+sidebar_label: "PDF preview"
+---
+---
+import PdfPreview from '/learn/assets/pdf-preview-and-download.png';
+
+Viewing PDFs without leaving the mobile app offers ability for developers to keep their users in the app's context. It is possible to let PDF file download to the phone and have the user open it using a different app. That may not be best user experience for an app. Not all users may know how to locate and open the PDF file in a different app. 
+In this blog we will see how to add PDF preview functionality to a WaveMaker app using the `react-native-pdf` and
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/react-native/pdf-preview-and-download.md:10]: In this 'how-to' or In this article

File: learn/react-native/pdf-preview-and-download.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+The `react-native-blob-util` library is a React Native wrapper for the `blob-util` library.
+We can leverage these libraries to add PDF preview functionality to a WaveMaker app. The best way to accomplish this is 
+through creating a prefab that will handle the PDF functional component.
+Please refer to this [page](https://docs.wavemaker.com/learn/react-native/custom-widget) for more information on how to 
+create a prefab that can be used in WaveMaker React Native project.
+
+
+## Downloading PDFs in a WaveMaker App
+
+Downloading PDFs is a common requirement in mobile applications, and React Native makes it relatively straightforward to implement this functionality.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/react-native/pdf-preview-and-download.md:26]: Does this work for PDFs which are streamed? for example, there is assumption here that all of the pdf content is avaialble in the viewer and blob content will give the file. Can the download action be bound to another REST API endpoint that triggers the same file download?

**Review by prasanth-reddy-wavemaker-com — CHANGES_REQUESTED** [review]

File: learn/react-native/pdf-preview-and-download.md (hunk 2 above)
Comment (vivek-dhayalan) [reply learn/react-native/pdf-preview-and-download.md:104]: @shilwantkumar can you please check this?

**Comment by vivek-dhayalan** [comment]
> mention

@shilwantkumar can you also please check password protected PDF?

**Comment by vivek-dhayalan** [comment]
> We should show how design the prefab with the PDF file url as a bindable parameter so that the mobile app developer who is using the prefab can program where the pdf file itself comes from.
> 
> We should also mention that the service serving the PDF file should be properly configured so that the preview works. For example, the service should not be setting 'Content-disposition' header to attachment.
> 
> Did we check if the preview works with pdf files that are password protected?

@shilwantkumar can you please check the password protected?

File: learn/react-native/pdf-preview-and-download.md (hunk 1 above)
Comment (shilwantkumar) [reply learn/react-native/pdf-preview-and-download.md]: @prasanth-reddy-wavemaker-com react-native-pdf supports password protected pdf file. Also, We will check if the preview works for the password protected PDF file or not.

Screenshot 2023-09-29 163604

File: learn/react-native/pdf-preview-and-download.md (hunk 2 above)
Comment (shilwantkumar) [reply learn/react-native/pdf-preview-and-download.md:104]: We are checking whether the preview works for password protected PDFs or not.

Comment (shilwantkumar) [reply learn/react-native/pdf-preview-and-download.md:126]: @prasanth-reddy-wavemaker-com we checked preview for  password protected pdf is working

**Comment by prasanth-reddy-wavemaker-com** [comment]
@swethakundaram why has this failed checks?

**Comment by swethakundaram** [comment]
> @swethakundaram why has this failed checks?

Cannot find this image file: '/learn/assets/bind-link-and-password-properties.png'

**Comment by shilwantkumar** [comment]
@prasanth-reddy-wavemaker-com added image '/learn/assets/bind-link-and-password-properties.png'

**Review by prasanth-reddy-wavemaker-com — APPROVED** [review]

**Pull request merged**